src/main/java/com/health/controller/RestApi.java

```diff
@@ -304,4 +345,114 @@ public ResponseEntity<Object> getContributorBoolean(@PathVariable(name = "tutori
+                resultMap.put(2, "No Enable Category exists for this catId: " + catId);
+                return ResponseEntity.ok(resultMap);
+            }
+        }
+
+        for (int lanId : uniquelanIds) {
+            Language lan = lanService.getById(lanId);
+            if (lan == null) {
+                resultMap.put(2, "No Language exists for this lanId: " + lanId);
+                return ResponseEntity.ok(resultMap);
+
+            }
+        }
+
+        String zipUrl = zipHealthTutorialThreadService.getZipName(courseName, uniqeCatIds, uniquelanIds, env);
+
+        if (zipUrl == null || zipUrl.isEmpty()) {
+            resultMap.put(0, "Zip creation in progress... Please check back after 30 minutes.");
+        } else if ("Error".equals(zipUrl)) {
+            resultMap.put(2, "No tutorials are available for the selected category and language.");
+        } else if (downloadCount.get() == downloadLimit) {
+            resultMap.put(0, "Download limit reached. Please try again after 30 minutes.");
+        } else {
+            String resultantzipUrl = ServiceUtility.convertFilePathToUrl(zipUrl);
+            resultMap.put(1, baseUrl + "/downloadManagerforhst?zipUrl=" + resultantzipUrl);
+
+        }
+
+        return ResponseEntity.ok(resultMap);
+    }
+
+    @GetMapping("/downloadManagerforhst")
+    public String downloadManager(@RequestParam(name = "zipUrl") String zipUrl, HttpServletResponse response) {
+        String message = "Please try again after 30 minutes.";
+        if (downloadCount.get() == downloadLimit) {
+
+            return message;
+        }
+        downloadCount.incrementAndGet();
+        logger.debug(" Increament downloadCount :{}", downloadCount.get());
+
+        Path zipFilePathName = Paths.get(env.getProperty("spring.applicationexternalPath.name"), zipUrl);
+        try (OutputStream os = new TimeoutOutputStream(response.getOutputStream(), downloadTimeOut);
+                InputStream is = new BufferedInputStream(new FileInputStream(zipFilePathName.toFile()));) {
+
+            response.setContentType(MediaType.APPLICATION_OCTET_STREAM_VALUE);
+            response.setHeader("Content-Disposition",
+                    "attachment; filename=" + ServiceUtility.getZipfileName(zipFilePathName));
+            response.setContentLengthLong(Files.size(zipFilePathName));
+
+            byte[] buffer = new byte[16384];
+            int length;
+
+            while ((length = is.read(buffer)) > 0) {
+
+                // TimeoutOutputStream tos = new TimeoutOutputStream(os, downloadTimeOut);
+                // tos.write(buffer, 0, length);
+
+                os.write(buffer, 0, length);
+
+            }
+            os.flush();
+
+        } catch (Exception e) {
+            logger.info("Exception:{}", e.getMessage());
+        } finally {
+            downloadCount.decrementAndGet();
+            logger.debug(" Decrement downloadCount :{}", downloadCount.get());
+        }
+
+        return null;
+
+    }
```

Comment:
you already have written download manager before. call the same function from here. do not repeat code.